### PR TITLE
Refactor Smile code

### PR DESCRIPTION
**Pull request by bouwew**
Implement EntityDescriptions

### CODE REVIEW

**Review by CoMPaTech — APPROVED** [review]

File: custom_components/plugwise/smile_helpers.py [hunk 1]
```diff
@@ -91,23 +91,19 @@ def _update_notify(self):
 
     def update_data(self):
         """Handle update callbacks."""
-        data = self._data[1][self._dev_id]
+        data = self._data[1][self._dev_id]["binary_sensors"]
 
-        for key, _ in data.items():
-            if key != "binary_sensors":
+        for _, item in enumerate(data):
```
Comment (CoMPaTech) [inline custom_components/plugwise/smile_helpers.py:96]: Would `for item in enumerate(data).items()` work?

File: custom_components/plugwise/smile_helpers.py [hunk 2]
```diff
@@ -91,23 +91,19 @@ def _update_notify(self):
...
-                    continue
-
-                self._is_on = item[ATTR_STATE]
-                self._icon = icon_selector(self._binary_sensor, self._is_on)
+            self._is_on = item[ATTR_STATE]
+            self._icon = icon_selector(self._binary_sensor, self._is_on)
 
-                if self._binary_sensor != "plugwise_notification":
-                    continue
+            if self._binary_sensor != "plugwise_notification":
```
Comment (CoMPaTech) [inline custom_components/plugwise/smile_helpers.py:103]: In less is more we could write as `if ==: update_notify` skipping the continue

**Review by CoMPaTech — COMMENTED** [review]
LG

**Review by brefra — APPROVED** [review]
As requested I've added a few commits to reduce/optimize the PlugwiseEntityDescriptions definition.

**Pull request merged**